src/helper.h

```diff
@@ -212,6 +215,9 @@ class SessionBase: public CbdataParent
     /// whether the server is locked for exclusive use by a client
     virtual bool reserved() = 0;
 
+    /// our creator (parent) object
+    virtual Client &helper() = 0;
+
```

Comment:
Now that we have helper(), please remove Helper::SessionBase::closePipesSafely() and closeWritePipeSafely() parameters.